core/node/state_keeper/src/io/persistence.rs

```diff
@@ -182,11 +180,6 @@ impl StateKeeperPersistence {
 
 #[async_trait]
 impl StateKeeperOutputHandler for StateKeeperPersistence {
-    async fn initialize(&mut self, cursor: &IoCursor) -> anyhow::Result<()> {
-        let mut connection = self.pool.connection_tagged("state_keeper").await?;
-        L2BlockSealProcess::clear_pending_l2_block(&mut connection, cursor.next_l2_block - 1).await
-    }
```

Comment:
AFAIU, `clear_pending_l2_block()` is always called earlier in both `StateKeeperIO::initialize()` implementations, so this call looks redundant.